iib/workers/tasks/build_recursive_related_bundles.py

```diff
@@ -0,0 +1,160 @@
+# The default value is 80. Set it to a more forgiving higher number to avoid issues
+yaml.width = 200
+# ruamel will also cause issues when normalizing a YAML object that contains
+# a nested JSON object when it does not preserve quotes. Thus, it produces
+# invalid YAML. Let's prevent this from happening at all.
+yaml.preserve_quotes = True
+log = logging.getLogger(__name__)
+
+
+@app.task
+@request_logger
+def handle_recursive_related_bundles_request(
+    parent_bundle_image: str,
+    organization: str,
+    request_id: int,
+    registry_auths: Optional[Dict[str, Any]] = None,
+) -> None:
+    """
+    Coordinate the work needed to find recursive related bundles of the operator bundle image.
+
+    :param str parent_bundle_image: the pull specification of the bundle image to whose related
+        bundles are to be found.
+    :param str organization: the name of the organization the bundle should be regenerated for.
+    :param int request_id: the ID of the IIB build request
+    :param dict registry_auths: Provide the dockerconfig.json for authentication to private
+      registries, defaults to ``None``.
+    :raises IIBError: if the recursive related bundles build fails.
+    """
+    _cleanup()
+
+    set_request_state(request_id, 'in_progress', 'Resolving parent_bundle_image')
+
+    with set_registry_auths(registry_auths):
+        parent_bundle_image_resolved = get_resolved_image(parent_bundle_image)
+
+        payload: UpdateRequestPayload = {
+            'parent_bundle_image_resolved': parent_bundle_image_resolved,
+            'state': 'in_progress',
+            'state_reason': (
+                f'Finding recursive related bundles for the bundle: {parent_bundle_image}'
+            ),
+        }
+        update_request(request_id, payload)
+
+        recursive_related_bundles = [parent_bundle_image_resolved]
+        current_level_related_bundles = [parent_bundle_image_resolved]
+        total_related_bundles = 0
+        conf = get_worker_config()
+        traversal_completed = False
+        while not traversal_completed:
+            temp_current_level_related_bundles = copy.deepcopy(current_level_related_bundles)
+            current_level_related_bundles = []
+            for bundle in temp_current_level_related_bundles:
+                children_related_bundles = process_parent_bundle_image(
+                    bundle, request_id, organization
+                )
+                current_level_related_bundles.extend(children_related_bundles)
+
+                total_related_bundles += len(children_related_bundles)
+                if total_related_bundles >= conf['iib_max_recursive_related_bundles']:
+                    raise IIBError('Max number of related bundles exceeded. Potential DDOS attack!')
+
+            recursive_related_bundles.extend(current_level_related_bundles)
+            if not current_level_related_bundles:
+                traversal_completed = True
```

Comment:
another way with less lines (no  traversal_completed) and not executing `.extend()` with empty dict might be:

```python
while True: 
    ...
    if not current_level_related_bundles:
        break

    recursive_related_bundles.extend(current_level_related_bundles)

```
I think I am ok with this one as well.

Reply:
You're right. The reason I went ahead with using the boolean was for readability. If you feel strongly that we would be better off if we don't use it then I can change it. Please let me know what you think :)